lib/fluent/plugin/in_systemd.rb

```diff
@@ -29,6 +29,7 @@ def start
       @running = true
       pos_writer.start
       @thread = Thread.new(&method(:run))
+      @thread.abort_on_exception = true
```

Comment:
It looks like the docs are wrong. I just tested this via
```ruby
#Thread.abort_on_exception = true
thread = Thread.new { sleep }
thread.abort_on_exception = true

thread2 = Thread.new { raise "Exception from thread" }
thread.join
```

If you uncomment the `Thread....` this will exit; otherwise this will hang. I opened an issue in the ruby bug tracker to fix the docs: https://bugs.ruby-lang.org/issues/12847